### PR TITLE
[OCPCLOUD-1226] add cluster-api-provider-alibaba repository

### PR DESCRIPTION
**Pull request by elmiko**
This change was created by running `make new-repo` and following the prompts.

### CODE REVIEW

**Comment by elmiko** [comment]
@Fedosin ptal, just want to make sure i did this correctly =)

File: ci-operator/config/openshift/cluster-api-provider-alibaba/openshift-cluster-api-provider-alibaba-main.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+binary_build_commands: make build
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.15
```
Comment (JoelSpeed) [inline ci-operator/config/openshift/cluster-api-provider-alibaba/openshift-cluster-api-provider-alibaba-main.yaml:6]:
```suggestion
    tag: golang-1.16
```

File: core-services/prow/02_config/openshift/cluster-api-provider-alibaba/_pluginconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+approve:
+- commandHelpLink: ""
+  repos:
+  - openshift/cluster-api-provider-alibaba
+  require_self_approval: false
```
Comment (JoelSpeed) [inline core-services/prow/02_config/openshift/cluster-api-provider-alibaba/_pluginconfig.yaml:5]: We tend to require self approval on our team's repos
```suggestion
  require_self_approval: true
```

**Comment by JoelSpeed** [comment]
> did you mean a file here or in the target repo?

IIRC there's some special configuration that has to go into this repo 🤔  @Fedosin you did this recently, have we got tide set up correctly?

**Comment by elmiko** [comment]
ack, i can review some of our other PRs

**Pull request closed without merging**